src/UserContext/AnonymousRequestMatcher.php

```diff
@@ -13,36 +13,62 @@
 
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * Matches anonymous requests using a list of identification headers.
  */
 class AnonymousRequestMatcher implements RequestMatcherInterface
 {
-    private $sessionNamePrefix;
+    /**
+     * The options configured in the constructor argument or default values.
+     *
+     * @var array
+     */
+    protected $options;
 
     /**
-     * @param string $sessionNamePrefix Prefix for session cookies. Must match your PHP session configuration
+     * When creating this matcher, you can configure a number of options.
+     *
+     * - user_identifier_headers: List of request headers that authenticate a non-anonymous request.
+     * - session_name_prefix:     Prefix for session cookies. Must match your PHP session configuration.
+     *
+     * @param array $options Options to overwrite the default options
+     *
+     * @throws \InvalidArgumentException if unknown keys are found in $options
      */
-    public function __construct($sessionNamePrefix)
+    public function __construct(array $options = array())
```

Comment:
i am afraid i already tagged a version with this class. please remove the type declaration and if $options happens to be a string, treat it as `['session_name_prefix' => $options]` to be BC.

Reply:
I can't see a tagged version with this class.

Reply:
oh, you are right, i missed to merge the 1.x branch to the master branch. so indeed, this is good as is. nice, was not happy that we would need a BC layer for something that has only just been added :-D